test/HostActivationTests/GivenThatICareAboutMultilevelSharedFxLookup.cs

```diff
@@ -0,0 +1,429 @@
+
+namespace Microsoft.DotNet.CoreSetup.Test.HostActivation.MultilevelSharedFxLookup
+{
+    public class GivenThatICareAboutMultilevelSharedFxLookup
+    {
+        private RepoDirectoriesProvider RepoDirectories;
+        private TestProjectFixture PreviouslyBuiltPortableTestProjectFixture;
+
+        private string _currentWorkingDir;
+        private string _userDir;
+        private string _executableDir;
+        private string _cwdSharedFxBaseDir;
+        private string _userSharedFxBaseDir;
+        private string _exeSharedFxBaseDir;
+        private string _builtSharedFxDir;
+        private string _cwdSelectedMessage;
+        private string _userSelectedMessage;
+        private string _exeSelectedMessage;
+        private string _sharedFxVersion;
+
+        public GivenThatICareAboutMultilevelSharedFxLookup()
+        {
+            // From the artifacts dir, it's possible to find where the sharedFrameworkPublish folder is. We need
+            // to locate it because we'll copy its contents into other folders
+            string artifactsDir = Environment.GetEnvironmentVariable("TEST_ARTIFACTS");
+            string builtDotnet = Path.Combine(artifactsDir, "..", "..", "intermediate", "sharedFrameworkPublish");
+
+            // The dotnetMultilevelSharedFxLookup dir will contain some folders and files that will be
+            // necessary to perform the tests
+            string multilevelDir = Path.Combine(artifactsDir, "dotnetMultilevelSharedFxLookup");
+
+            // The three tested locations will be the cwd, the user folder and the exe dir. Both cwd and exe dir
+            // are easily overwritten, so they will be placed inside the multilevel folder. The actual user location will
+            // be used during tests
+            _currentWorkingDir = Path.Combine(multilevelDir, "cwd");
+            if (RuntimeEnvironment.OperatingSystemPlatform == Platform.Windows)
+            {
+                _userDir = Environment.GetEnvironmentVariable("USERPROFILE");
+            }
+            else
+            {
+                _userDir = Environment.GetEnvironmentVariable("HOME");
+            }
+            _executableDir = Path.Combine(multilevelDir, "exe");
+
+            // SharedFxBaseDirs contain all available version folders
+            _cwdSharedFxBaseDir = Path.Combine(_currentWorkingDir, "shared", "Microsoft.NETCore.App");
+            _userSharedFxBaseDir = Path.Combine(_userDir, ".dotnet", RuntimeEnvironment.RuntimeArchitecture, "shared", "Microsoft.NETCore.App");
+            _exeSharedFxBaseDir = Path.Combine(_executableDir, "shared", "Microsoft.NETCore.App");
+
+            // Completely remove remaining files from other tests. Actually, if the tests were successfully
+            // completed, the folder is supposed to be empty. We're just playing safe
+            if (Directory.Exists(multilevelDir))
+            {
+                Directory.Delete(multilevelDir, true);
```

Comment:
@gkhanna79 The other tests fail here, since this is the first time Directory.Delete is called for them (multilevelDir didn't exist during the first test).